tests/test_automatic_fields.py

```diff
@@ -15,38 +21,49 @@ def test_run(self):
         conn_id = connections.ensure_connection(self)
         expected_streams = self.expected_streams()
 
-        # run check mode
+        # Run check mode
         found_catalogs = self.run_and_verify_check_mode(conn_id)
 
-        # de-select all the fields
-        self.select_found_catalogs(conn_id, found_catalogs, deselect_all_fields=True, only_streams=expected_streams)
+        # De-select all the fields
+        self.select_found_catalogs(conn_id, found_catalogs,
+                                   deselect_all_fields=True, only_streams=expected_streams)
 
-        # run sync
+        # Run sync
         record_count_by_stream = self.run_and_verify_sync(conn_id)
         synced_records = runner.get_records_from_target_output()
 
         for stream in expected_streams:
             with self.subTest(stream=stream):
 
-                # expected values
+                # Expected values
                 expected_primary_keys = self.expected_primary_keys()[stream]
-                expected_keys = expected_primary_keys | self.expected_replication_keys()[stream]
+                expected_automatic_keys = expected_primary_keys | self.expected_replication_keys()[stream]
 
-                # collect actual values
+                # Collect actual values
                 messages = synced_records.get(stream)
-                record_messages_keys = [set(row['data'].keys()) for row in messages['messages']]
+                record_messages_keys = [set(row['data'].keys())
+                                        for row in messages['messages']]
 
-                # check if the stream has collected some records
+                # Check if the stream has collected some records
                 self.assertGreater(record_count_by_stream.get(stream, 0), 0)
 
                 # Verify that only the automatic fields are sent to the target
                 for actual_keys in record_messages_keys:
-                    self.assertSetEqual(expected_keys, actual_keys)
+                    self.assertSetEqual(expected_automatic_keys, actual_keys)
+
+                # Get records
+                records = [message.get("data") for message in messages.get('messages', [])
+                           if message.get('action') == 'upsert']
+
+                # Remove duplicate records
+                records_pks_list = [tuple(message.get(pk) for pk in expected_primary_keys)
+                                   for message in [json.loads(t) for t in {json.dumps(d) for d in records}]]
+
+                # Remove duplicate primary keys
+                records_pks_set = set(records_pks_list)
```

Comment:
Can you please write appropriate names instead of `records_pks_list` and `records_pks_set`, these 2 sound too conceptual.